plugins/hls-class-plugin/src/Ide/Plugin/Class.hs

```diff
@@ -234,18 +253,55 @@ codeAction state plId (CodeActionParams _ _ docId _ context) = liftIO $ fmap (fr
           _ -> panic "Ide.Plugin.Class.findClassFromIdentifier"
     findClassFromIdentifier _ (Left _) = panic "Ide.Plugin.Class.findClassIdentifier"
 
+    findImplementedMethods :: NormalizedFilePath -> Range -> MaybeT IO [T.Text]
+    findImplementedMethods docPath range = do
+        (HAR {hieAst = hf}, pmap) <- MaybeT . runAction "classplugin" state $ useWithStale GetHieAst docPath
```

Comment:
Both this and `findClassIdentifier` run an action to get the HIE AST. I'm not 100% sure, but I suspect it may be more efficient to only do that once. Perhaps pull the action out of the two functions and just pass in the HIE ast to them?

Reply:
Also, they both run a `pointCommand` over the same range. Maybe not worth the refactoring, but in principle you could just run one that returns both the things we're interested in.